### ISSUE TITLE
Separate tests into a `scipy-tests` package

### ISSUE THREAD

**Issue by xhochy** [conda-forge/scipy-feedstock#160]
- [x] I read [the conda-forge documentation](https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge) and could not find the solution for my problem there.

The tests in the scipy package are quite large (11MiB on-disk size). Thus it would be nice if we could ship them in a separate conda package like we did for `pyarrow`: https://github.com/conda-forge/arrow-cpp-feedstock/blob/48de7951f41b685cd1112e2f85c7c016eac0f553/recipe/meta.yaml#L232 This would allow downstream consumers to still run the `scipy` tests in their environment but allow those that don't want to do this to not ship the tests at all.

**Comment by rgommers** [comment]
If I understand the `SP_DIR` bit correctly, the separate `arrow-tests` package installs files into the same location as where the files for `arrow` are, not in a separate top-level directory. Is that correct? If so, separating seems doable. Otherwise it would be a fair bit of surgery, e.g. there's extensive use of `dirname(__file__)` in the tests.

**Comment by jakirkham** [comment]
Yeah exactly. Split packages are just a way to divide up what was installed into smaller pieces

Agree moving files around would complicate things unnecessarily

**Comment by rgommers** [comment]
Assuming this works, sounds fine to me. We'll have to document this and perhaps adjust `scipy._lib._testutils.PytestTester` to ensure it behaves nicely if someone calls `scipy.test()` or `scipy.some_submodule.test()`.

**Comment by xhochy** [comment]
Should these simply return an error with the information to install scipy-tests in that case?

**Comment by xhochy** [comment]
Should these simply return an error with the information to install scipy-tests in that case?

**Comment by rgommers** [comment]
Yes that sounds good.
